packages/legend-extension-dsl-data-space/src/components/query/DataSpaceQueryBuilderTemplateQueryPanelContent.tsx

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  PanelHeader,
+  FilterIcon,
+  BasePopover,
+  ClickAwayListener,
+  ShareIcon,
+  TagIcon,
+} from '@finos/legend-art';
+import { observer } from 'mobx-react-lite';
+import { useRef } from 'react';
+import { DataSpaceExecutableTemplate } from '../../graph/metamodel/pure/model/packageableElements/dataSpace/DSL_DataSpace_DataSpace.js';
+import type { DataSpaceQueryBuilderState } from '../../stores/query/DataSpaceQueryBuilderState.js';
+import { useApplicationStore } from '@finos/legend-application';
+import { generateDataSpaceTemplateQueryCreatorRoute } from '../../__lib__/query/DSL_DataSpace_LegendQueryNavigation.js';
+
+const DataSpaceTemplateQueryDialog = observer(
+  (props: {
+    triggerElement: HTMLElement | null;
+    queryBuilderState: DataSpaceQueryBuilderState;
+    templateQueries: DataSpaceExecutableTemplate[];
+  }) => {
+    const { triggerElement, queryBuilderState, templateQueries } = props;
+    const applicationStore = useApplicationStore();
+    const handleClose = (): void => {
+      queryBuilderState.setTemplateQueryDialogOpen(false);
+    };
+    const loadQuery = (template: DataSpaceExecutableTemplate): void => {
+      const executionContext =
```

Comment:
this works if we are running this in an existing query as well right? 
If the existing query is coming from a dataSpace?
have we thought about adding a confirm modal if the query has unsaved changes in an existing query as they would lose they change?

Reply:
yes it will work 
I added the alert 

https://github.com/finos/legend-studio/assets/73408381/0854effe-c339-4f60-b28d-e6dcb85c1961